app/concerns/chooser_endpoints.rb

```diff
@@ -45,7 +45,8 @@ def get_provider_collections(params = {})
     collection_params['keyword'].concat('*') if collection_params.key?('keyword')
 
     # Retreive the collections from CMR, allowing a few additional parameters
-    cmr_client.get_collections_by_post(collection_params, token).body
+    response = cmr_client.get_collections_by_post(collection_params, token)
+    response.body if response.success?
```

Comment:
It seems like we should be providing a response body if the response is an `error?` as well. Here it should probably be a `{}` because the in some cases `render_collections_for_chooser` method above will be called, passing in what is returned from this method.